### ISSUE TITLE
Interpolating expression with shorthand if-else return unexpected value

### ISSUE THREAD

**Issue by tom10271** [angular/angular#10839]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
`{{ 'HB.height_modifier.' + (true ? 'expand' : 'collapse') }}` returns `expand`

**Expected/desired behavior**
`'HB.height_modifier.' + (true ? 'expand' : 'collapse')` should return `HB.height_modifier.expand` as tested in browser console

**Please tell us about your environment:**

* **Angular version:** 2.0.0-beta.16
 
* **Browser:** Chrome 52 on Mac OSX
* **Language:** TS 1.x transpiled with SystemJS(I think)

**Comment by tom10271** [comment]
Of course, I can write `{{ !state.trending.value ? 'HB.height_modifier.expand' : 'HB.height_modifier.collapse' }}` but `{{ 'HB.height_modifier.' + !state.trending.value ? 'expand' : 'collapse' }}` is much better IMO

**Comment by pkozlowski-opensource** [comment]
> 2.0.0-beta.16

Uhh, this is like 4 months old version. The issue you are reporting was fixed in maser via 1ad2a02b111629c44be6027ffcbc70de440bb734 and the fix is available from 2.0.0-beta.17 onward.

Closing as already fixed in master.
